phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/LibraryFilter.java

```diff
@@ -0,0 +1,98 @@
+package org.phoenicis.javafx.views.mainwindow.library;
+
+import org.phoenicis.javafx.views.AbstractFilter;
+import org.phoenicis.library.dto.ShortcutCategoryDTO;
+import org.phoenicis.library.dto.ShortcutDTO;
+
+import java.util.Optional;
+
+/**
+ * Filter class for the "Library" tab
+ *
+ * @author Marc Arndt
+ */
+public class LibraryFilter extends AbstractFilter {
+    /**
+     * The entered search term.
+     * If no search term has been entered, this value is {@link Optional#empty()}.
+     */
+    private Optional<String> searchTerm;
+
+    /**
+     * The selected shortcut category.
+     * If no shortcut category has been selected, this value is {@link Optional#empty()}.
+     */
+    private Optional<ShortcutCategoryDTO> selectedShortcutCategory;
```

Comment:
Isn't this really the same as for Installations? Would it be possible to do something like:
```Java
public class FilterWithCategory<T> extends AbstractFilter {
...
    private Optioinal<T> selectedCategory;
```

Reply:
The only similar thing are the fields (and method names) inside the classes.
The contents of the filter methods differs, if only in one word.